### PR TITLE
Reorganization of functions -- prep for operational version

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses operational roadmap and #271.

Some functions need to be moved about to facilitate the breakdown into a core pysat package and several toolboxes.

Summary of Changes
  - Move `computational_form` to `ssnl`
  - Move `scale_units` to `utils._core`
  - Rename `season_date_range` as `create_date_range`
  - Added deprecation warnings to stat functions
  - Reorganized relevant unit tests

https://github.com/rstoneback/pysat/wiki/Proposed-Division-of-pysat-functions
https://github.com/rstoneback/pysat/wiki/Proposed-Reorganization-Meeting-August-28,-2019

## Type of change

- Breaking change (fix or feature that would cause existing functionality
      to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

Unit Tests on Travis CI

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [na] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/ssnl/__init__.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 """
 
-from . import occur_prob
-from . import avg
-from . import plot
+from . import occur_prob  # noqa 401
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/__init__.py]: Ok, best practice would recommend ensuring all comments are human readable.

[the remark this answers is not in the document]
Comment (aburrell) [reply pysat/ssnl/__init__.py]: The `noqa 401` also occurs elsewhere, I didn't flag all of them.

[the remark this answers is not in the document]
Comment (jklenzing) [reply pysat/ssnl/__init__.py]: pulled these out for now

[the remark this answers is not in the document]
Comment (rstoneback) [reply pysat/ssnl/__init__.py]: @aburrell Thanks for keeping standards high.

**Comment by jklenzing** [comment]
Haven't merged this as a new bug has popped in the the download tests for sw_f107.  It looks like this is popping up in #264 as well, so it's possible the data repository has changed something.

**Comment by jklenzing** [comment]
Verified in an old branch which now fails.  Documenting.

**Pull request merged**